operator/v1/types_clustermanager.go

```diff
@@ -132,8 +132,119 @@ type RegistrationDriverHub struct {
+	Port int `json:"port,omitempty"`
+	// Name is the name of the database to connect to.
+	Name string `json:"name,omitempty"`
+	// Username is the username to use for authentication.
+	Username string `json:"username,omitempty"`
+	// Password is the password to use for authentication.
+	Password string `json:"password,omitempty"`
+	// SSLMode specifies the SSL mode for the database connection.
+	SSLMode string `json:"sslmode,omitempty"`
+	// CACertFile is the path to the CA certificate file used for SSL connections.
+	CACertFile string `json:"ca_cert_file,omitempty"`
 }
 
+// GRPC represents the configuration for gRPC driver.
+type GRPC struct {
+	// GRPCConfigSecretName is the name of the secret that contains the gRPC broker configuration.
+	// if the secret is not provided, the clustermanager will sign the certificates for gRPC broker.
+	// The secret should contain config.yaml, client-ca.crt, client-ca.key, server.crt, server.key, and root-ca.crt.
+	// The config.yaml should contain the following keys:
+	// - client_ca_file: The path to the CA certificate file used for client authentication.
+	// - client_ca_key_file: The path to the CA key file used for client authentication.
+	// - server_cert_file: The path to the server certificate file used for gRPC server.
+	// - server_key_file: The path to the server key file used for gRPC server.
+	// - root_ca_file: The path to the root CA certificate file used for client authentication.
+	// client-ca.crt, client-ca.key, server.crt, server.key, and root-ca.crt should be in PEM format.
+	// +optional
+	GRPCConfigSecretName string `json:"name,omitempty"`
+
+	// DBConfigSecretName is the name of the secret that contains the database configuration for gRPC broker.
+	// The secret should contain config.yaml, ca.crt.
+	// The config.yaml should contain the following keys:
+	// - host: The hostname or IP address of the database server.
+	// - port: The port number on which the database server is listening.
+	// - name: The name of the database to connect to.
+	// - username: The username to use for authentication.
+	// - password: The password to use for authentication.
+	// - sslmode: The SSL mode for the database connection.
+	// - certificate_file: The path to the CA certificate file used for SSL connections.
+	// The ca.crt should be in PEM format.
+	// +optional
+	DBConfigSecretName string `json:"dbConfigSecretName,omitempty"`
+
+	// ImagePullSpec represents the desired image of the gRPC broker installed on hub.
+	// +optional
+	// +kubebuilder:default=quay.io/open-cluster-management/grpc-broker
+	ImagePullSpec string `json:"imagePullSpec,omitempty"`
+
+	// EndpointPublishingType represents the type of endpoint publishing for gRPC.
+	// TODO: support loadbalancer and ingress.
+	// +required
+	// +kubebuilder:default:=nodeport
+	// +kubebuilder:validation:Enum=nodeport;route
+	EndpointPublishingType GRPCEndpointPublishingType `json:"endpointPublishingType,omitempty"`
```

Comment:
EndpointPublishingType and the following should be in a nested struct. Also we should have loadbalancer, gateway as the types.